Add files via upload

AnastasiiaRazheva opened the pull request:
Test comit. 
Aadded a short description of the future module in ToolButtonList.prototype.help

AnastasiiaRazheva committed:
Add files via upload

Test comit. 
Aadded a short description of the future module in ToolButtonList.prototype.help

eakarpov reviewed (changes requested):
Обнови сразу зависимость ядра до 1.0.12 и поменяй импорты:
TaskTemplate -> Template
TaskToolbar -> Toolbar

AnastasiiaRazheva commented:
npm v graphlabs.core.template
dist-tags: latest: 1.0.16

Насчёт TaskTemplate -> Template
import Template
Модуль ""../../../../../../../../Users/Russi/OneDrive/\u0414\u043E\u043A\u0443\u043C\u0435\u043D\u0442\u044B/\u0423\u0418\u0420/0.1/Tasks.Planarity-master/node_modules/graphlabs.core.template/build"" не имеет экспортированного элемента "Template".ts(2305)

AnastasiiaRazheva committed:
Fail try to use DFS

Исправила Замечания по синтаксису.

Пыталась подключить для поиска циклов функцию dfs - но это не так просто. Пока оставлю так.

AnastasiiaRazheva committed:
It work!

Я все выходные на это убила, но оно начало работать.
Интерфейс и оптимизация кода завтра.

Commit:
radio button check

AnastasiiaRazheva committed:
reName + radiobutton

Удалила отладочные строки.
Переименовала, чтобы было без "_".
Добавила RadioButton - пока толком не работают ((.

AnastasiiaRazheva committed:
OOP + splitGraphByBridge

Перенесла всё в класс BetaGraph.
Добавила разбиение Графа в случае наличия моста.

AnastasiiaRazheva committed:
Merge branch 'AnastasiiaRazheva-patch-1' of https://github.com/GraphLabsSuite/Tasks.Planarity into AnastasiiaRazheva-patch-1

AnastasiiaRazheva committed:
build + componentDidMount

Я перенесла проверку планарности в componentDidMount

eakarpov commented on src/Segment.tsx:
```diff
@@ -0,0 +1,26 @@
+
+export class Segment {
+
+    bodyEdges: IEdgeView[] = [];
```
где модификаторы доступа все? непонятно, что можно использовать что нет снаружи

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+            this.findSegments();
+        }
+    
+        if (this.segments.length == 0){
```
===

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+
+        this.vertices = this.vertices.filter((v: IVertexView) => this.edges.find((e: IEdgeView) => e.vertexOne == v.name || e.vertexTwo == v.name ));
+    
+        var numVerticesDeleted: number = 0;
```
var думаю здесь не оправдан

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+            
+            this.edges = this.edges.filter((e: IEdgeView) => this.vertices.find((v: IVertexView) => v.name === e.vertexOne ) && this.vertices.find((v: IVertexView) => v.name === e.vertexTwo ));
+            
+        } while(numVerticesDeleted != 0);
```
!==

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+        this.edges.filter(nonUsedEdge => !this.usedEdges.find(e => e == nonUsedEdge)).forEach(nonUsedEdge => {
+            if (this.usedVertices.find(v => nonUsedEdge.vertexOne == v.name) && this.usedVertices.find(v => nonUsedEdge.vertexTwo == v.name)){
+                this.segments.push(new Segment([nonUsedEdge], 
+                                                [this.vertices.find(v => v.name == nonUsedEdge.vertexOne), 
```
может сделать просто filter чем четыре раза проходиться по массиву?

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+            } else 
+            {
+                resultCycle.vertices = resultCycle.vertices.concat([this.vertices.find((v: IVertexView) => v.name === nextEdge.vertexOne)]);
+                colored = colored.concat([this.vertices.find((v: IVertexView) => v.name === nextEdge.vertexOne)]);
```
filter лучше опять-таки

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+            this.usedVertices = this.usedVertices.concat(minValueSegment.bodyVertices.slice(1, minValueSegment.bodyVertices.length - 1));
+            this.usedEdges = this.usedEdges.concat(minValueSegment.bodyEdges.slice());
+    
+            var newVertices1 : IVertexView[];
```
let variable: Type;
и что за newVertices1? Какой подход к нумерации? Комментарии или переименовать.

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+        
+        this.findSegments();
+
+        var counter = 0;
```
let

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+
+        var counter = 0;
+        
+        while (this.segments.length != 0 && !this.segments.find(s => s.value == 0) && counter < 100){
```
!==, ===

eakarpov commented on src/BetaGraph.tsx:
```diff
@@ -0,0 +1,307 @@
+        var resultSegment: Segment = new Segment([], [startPoint], startPoint, null);
+        let blackList: IEdgeView[] = [];
+
+        let nameOfBlockedVertex: string = null;
```
предполагаю что тип string|null

AnastasiiaRazheva committed:
Hot fix

Исправление замечаний по результату ревью от Егора Карпова

AnastasiiaRazheva committed:
Летняя практика.

Код для будущего оценивания ответов студентов. В рамках летней практики.

AnastasiiaRazheva committed:
npm update + log

Обновила шаблон и запилила логи

AnastasiiaRazheva committed:
Change graphlabs.core.template version by hand

AnastasiiaRazheva committed:
rename TaskTemplate and another ....

AnastasiiaRazheva committed:
disable logs.

AnastasiiaRazheva committed:
Стили та подгрузила

AnastasiiaRazheva committed:
Поддержала обновления core.template

Обновила template до 1.0.32
Добавила graphlabs.core.graphs

svtz committed:
update template

AnastasiiaRazheva committed:
Поддержала переход на graphModel и обновила graphs и template

AnastasiiaRazheva committed:
Merge branch 'AnastasiiaRazheva-patch-1' of https://github.com/GraphLabsSuite/Tasks.Planarity into AnastasiiaRazheva-patch-1
